app/assets/javascripts/webchat/library.js

```diff
@@ -27,14 +23,19 @@
       if (!availabilityUrl || !openUrl) {
         throw Error.new('urls for webchat not defined')
       }
-      $openButton.on('click', handleOpenChat)
+
+      if (redirectUrl === true) {
+        var openButton = document.getElementsByClassName('.js-webchat-open-button')[0]
+        openButton.addEventListener('click', handleOpenChat)
+      }
       intervalID = setInterval(checkAvailability, POLL_INTERVAL)
       checkAvailability()
     }
 
     function handleOpenChat (evt) {
       evt.preventDefault()
-      this.dataset.redirect === 'true' ? window.location.href = openUrl : global.open(openUrl, 'newwin', 'width=366,height=516')
+      var redirect = this.getAttribute('data-redirect')
+      redirect === 'true' ? window.location.href = openUrl : window.open(openUrl, 'newwin', 'width=366,height=516')
```

Comment:
I'm not certain, but I think this line isn't the equivalent of the original. In the original, `this.dataset.redirect` is set to something based on the ternary operator. In the new code, only the `redirect` variable is updated.

I think this might need a line adding after it to set the attribute back onto the element, e.g.

```
redirect === 'true' ? window.location.href = openUrl : window.open(openUrl, 'newwin', 'width=366,height=516')
this.setAttribute('data-redirect', redirect)
```

Having said that, I'm not sure how changing this attribute would change anything, since the only instance of it being called is once when the JS is initialised. So, confused.

Reply:
Oh thanks! I created the `redirect` variable because otherwise the line length made the statement unreadable. But yes, it makes sense that the original element needs to be updated, thanks!

And yes, I'm confused too!